src/firebolt/client/auth.py

```diff
@@ -12,16 +12,26 @@
 
 
 class Auth(HttpxAuth):
+    """Authentication class for Firebolt database.
 
-    """
-    Authentication class for Firebolt database. Gets authentication token using
+    Gets authentication token using
     provided credentials and updates it when it expires
+
+    Args:
+        username (str): Username
+        password (str): Password
+        api_endpoint (Optional[str]): Environment api endpoint.
+            Default api.app.firebolt.io
+
+    Attributes:
+        username (str): Username
+        password (str): Password
     """
 
     __slots__ = (
         "username",
         "password",
-        "api_url",
+        "_api_endpoint",
```

Comment:
Love that we spotted this! However, I thought it would fail on runtime if an instance variable was not declared in __slots__ ? How was this working before?

Reply:
This is actually a tricky one.
The reason seems to be that base Auth class doesn't have `__slots__` defined, and therefore it creates a `__dict__` entry, which makes `__slots__` redundant. I think in this case we can leave it just for documentation purposes